datafusion/expr/src/built_in_function.rs

```diff
@@ -890,7 +890,7 @@ impl BuiltinScalarFunction {
                 // 0 or more arguments of arbitrary type
                 Signature::one_of(vec![VariadicAny, Any(0)], self.volatility())
             }
-            BuiltinScalarFunction::Struct => Signature::variadic(
+            BuiltinScalarFunction::Struct => Signature::variadic_limited(
```

Comment:
What do you think about using `Signature::variadic_any`? That I think would pass along the argument types directly (not try to coerce them)